analysis/Analysis/Section_3_4.lean

```diff
@@ -397,6 +397,14 @@ lemma SetTheory.Set.mem_powerset' {S S' : Set} : (S': Object) ∈ S.powerset ↔
   rw [mem_powerset]
   simp
 
+/-- Another helper lemma for Exercise 3.4.7. -/
+@[simp]
```

Comment:
Maybe this shouldn't be `@[simp]`

With simp sets, you want confluence, which is the property that no matter which order simp lemmas are applied, eventually simp will normalize an expression to the same thing. If you need `tauto` then that suggests that this lemma, `union_axiom`, and `replacement_axiom` can't all be in the same simp set.